spec/domain/operations/individual/fetch_existing_coverage_spec.rb

```diff
@@ -0,0 +1,163 @@
+                        product: product,
+                        household: family.active_household,
+                        coverage_kind: "health",
+                        effective_on: effective_on + 6.months,
+                        terminated_on: effective_on.end_of_year,
+                        kind: 'employer_sponsored',
+                        hbx_enrollment_members: [hbx_en_member1, hbx_en_member2],
+                        aasm_state: 'coverage_selected')
+    end
+
+    let!(:enrollment5) do
+      FactoryBot.create(:hbx_enrollment,
+                        family: family,
+                        product: product2,
+                        household: family.active_household,
+                        coverage_kind: "dental",
+                        effective_on: effective_on,
+                        terminated_on: effective_on.next_month.end_of_month,
+                        kind: 'individual',
+                        hbx_enrollment_members: [hbx_en_member1, hbx_en_member2],
+                        aasm_state: 'coverage_selected')
+    end
+    let!(:enrollment6) do
+      FactoryBot.create(:hbx_enrollment,
+                        family: family,
+                        product: product2,
+                        kind: 'individual',
+                        household: family.active_household,
+                        coverage_kind: "dental",
+                        hbx_enrollment_members: [hbx_en_member2],
+                        effective_on: effective_on + 2.months,
+                        terminated_on: (effective_on + 5.months).end_of_month,
+                        aasm_state: 'shopping')
+    end
+
+    let(:params) do
+      { :enrollment_id => enrollment2.id }
+    end
+
+    context "Not passing params to call the operation" do
+      let(:params) { { } }
+
+      it "fails" do
+        expect(subject).not_to be_success
+        expect(subject.failure).to eq "Given input is not a valid enrollment id"
+      end
+    end
+
+    context "passing incorrect params to call the operation" do
+      let(:params) { { :enrollment_id => 'abc123' } }
+
+      it "fails" do
+        expect(subject).not_to be_success
+        expect(subject.failure).to eq "Given input is not a valid enrollment id"
+      end
+    end
+
+    context "passing enrollment with no hbx enrollment members" do
+      before :each do
+        enrollment2.hbx_enrollment_members.destroy_all
+        enrollment2.save!
+      end
+
+      it "returns failure" do
+        expect(subject).not_to be_success
+        expect(subject.failure).to eq "Enrollment does not include dependents"
+      end
+    end
+
+    context "hbx enrollment members with shopping enrollment" do
+      it "returns success" do
+        expect(subject).to be_success
+        expect(subject.success).to eq [enrollment1, enrollment3]
+        # SHOP Enrollment not included
+        expect(subject.success).not_to include [enrollment4]
+      end
+    end
+
```

Comment:
Need more scenarios to check the market_kind, coverage_kind, what is the enrollment as no dependents, and if the enrollments are missing family_ids.